micrometer/prometheus/src/main/java/io/quarkus/ts/openshift/micrometer/PrimeNumberResource.java

```diff
@@ -0,0 +1,76 @@
+package io.quarkus.ts.openshift.micrometer;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import org.apache.commons.lang3.RandomStringUtils;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import java.util.concurrent.atomic.LongAccumulator;
+import java.util.function.Supplier;
+
+@Path("/")
+public class PrimeNumberResource {
+
+    private final LongAccumulator highestPrime = new LongAccumulator(Long::max, 0);
+    private final MeterRegistry registry;
+    private final String uniqueId = RandomStringUtils.randomAlphabetic(5);
+
+    PrimeNumberResource(MeterRegistry registry) {
+        this.registry = registry;
+
+        // Create a gauge that uses the highestPrimeNumberSoFar method
+        // to obtain the highest observed prime number
+        registry.gauge("prime.number.max." + uniqueId, this,
```

Comment:
Not sure if its a good idea to create a metric with a Unique Key.  I will suggest use as a key "prime.number.max" and then add tags . So, in this way, we could make queries over "prime.number.max"  and then filter by tag.

Reply:
Metrics will be deleted after some time from Prometheus. 

Using a unique id will extremely ease the test verification (see how we assert the metrics).